### PR TITLE
prov/gni: switch to using mntent interface

### PR DESCRIPTION
**Pull request by hppritcha**
With CLE 6.x, the entries in /proc/mounts can be quite
large.  Use the mntent interfaces to parse /proc/mounts
to avoid buffer overflows.

@ztiffany 
@sungeunchoi 

Signed-off-by: Howard Pritchard howardp@lanl.gov

### CODE REVIEW

File: prov/gni/src/gnix_mbox_allocator.c [hunk 1]
```diff
@@ -51,74 +53,45 @@
...
-	if (!fd) {
-		error = strerror_r(errno, error_buf, sizeof(error_buf));
-		GNIX_WARN(FI_LOG_EP_CTRL, "Error opening /proc/mounts: %s\n",
-			  error);
-		bytes_read = -FI_EIO;
-		goto err_fopen;
+	fd = setmntent ("/proc/mounts", "r");
+	if (fd == NULL) {
+		GNIX_WARN(FI_LOG_EP_CTRL,
+			  "Invalid page_size or directory provided.\n");
```
Comment (sungeunchoi) [inline prov/gni/src/gnix_mbox_allocator.c]: This error message doesn't seem quite right.  It's probably just fine to use the original message.

**Comment by sungeunchoi** [comment]
👍  I like this better regardless of CLE version.

**Comment by hppritcha** [comment]
I'll fix the comment.

**Comment by hppritcha** [comment]
comment fixed

**Comment by ztiffany** [comment]
cool! :+1:

**Pull request merged**